tastypie/resources.py

```diff
@@ -834,9 +834,31 @@ def full_dehydrate(self, bundle, for_list=False):
         to populate the resource.
         """
         use_in = ['all', 'list' if for_list else 'detail']
+        
+        all_fields = set(self.fields.keys())
+
+        allowed_fields = None
+        excluded_fields = None
+        
+        if not bundle.related_obj:
+            if 'fields' in bundle.request.GET:
+                allowed_fields = bundle.request.GET.getlist('fields')
```

Comment:
instead of requiring `fields` to be respecified over and over, just use comma separation.